dialoguekit/nlg/nlg_template.py

```diff
@@ -100,22 +106,30 @@ def generate_utterance_text(
         response_utterance = random.choice(templates)
         response_utterance = deepcopy(response_utterance)
 
-        # Clear out annotations
+        # Clear out dialogue acts and annotations
+        response_utterance.dialogue_acts = []
         response_utterance.annotations = []
 
-        if annotations is not None:
-            for annotation in annotations:
-                response_utterance.text = response_utterance.text.replace(
-                    f"{{{annotation.slot}}}", annotation.value, 1
-                )
-            response_utterance.add_annotations(annotations)
+        # If annotations are provided, use them to fill in the template
+        for da in dialogue_acts:
+            if da.annotations:
+                for annotation in da.annotations:
+                    response_utterance.text = response_utterance.text.replace(
+                        f"{{{annotation.slot}}}", annotation.value, 1
+                    )
+            response_utterance.add_dialogue_acts([da])
+        for annotation in annotations:
+            response_utterance.text = response_utterance.text.replace(
+                f"{{{annotation.slot}}}", annotation.value, 1
+            )
+            response_utterance.add_annotations([annotation])
```

Comment:
This part seems to be exactly the same in `nlg_conditional.py`. Consider factoring it out (under nlg?) to avoid duplication.